CMakeLists.txt

```diff
@@ -633,24 +583,133 @@ if (WIN32)
 			_WIN32_WINNT=0x0600 # target Windows Vista or later
 			WIN32_LEAN_AND_MEAN # prevent winsock1 to be included
 	)
+
 	if (MSVC)
-		target_compile_definitions(torrent-rasterbar PUBLIC BOOST_ALL_NO_LIB)
+		target_compile_definitions(torrent-rasterbar
+			PUBLIC
+				BOOST_ALL_NO_LIB
+				_SCL_SECURE_NO_DEPRECATE _CRT_SECURE_NO_DEPRECATE # disable bogus deprecation warnings on msvc8
+		)
+		target_compile_options(torrent-rasterbar
+			PRIVATE
+				/Zc:wchar_t /Zc:forScope # these compiler settings just make the compiler standard conforming
+				/MP # for multi-core compilation
+				/bigobj # increase the number of sections for obj files
+		)
+		set_target_properties(torrent-rasterbar PROPERTIES LINK_FLAGS_RELEASE "/OPT:ICF=5 /OPT:REF")
 	endif()
 endif()
 
-if(ANDROID)
-    target_link_libraries(torrent-rasterbar PRIVATE ${CMAKE_DL_LIBS})
+if (ANDROID)
+	target_link_libraries(torrent-rasterbar PRIVATE ${CMAKE_DL_LIBS})
 endif()
 
-if(APPLE)
+if (APPLE)
 	# for ip_notifier
 	target_link_libraries(torrent-rasterbar PRIVATE "-framework CoreFoundation" "-framework SystemConfiguration")
-endif (APPLE)
+endif()
+
+feature_option(build_tests "build tests" OFF)
+feature_option(build_examples "build examples" OFF)
+feature_option(build_tools "build tools" OFF)
+feature_option(python-bindings "build python bindings" OFF)
+
+# these options require existing target
+feature_option(dht "enable support for Mainline DHT" ON)
+if(NOT build_tests) # tests require deprecated symbols
+	target_optional_compile_definitions(torrent-rasterbar PUBLIC FEATURE NAME deprecated-functions DEFAULT ON
+		DESCRIPTION "enable deprecated functions for backwards compatibility" DISABLED TORRENT_NO_DEPRECATE)
+endif()
+feature_option(encryption "Enables encryption in libtorrent" ON)
+feature_option(exceptions "build with exception support" ON)
+target_optional_compile_definitions(torrent-rasterbar PUBLIC FEATURE NAME extensions DEFAULT ON
+	DESCRIPTION "Enables protocol extensions" DISABLED TORRENT_DISABLE_EXTENSIONS
+)
+target_optional_compile_definitions(torrent-rasterbar PUBLIC FEATURE NAME extensions DEFAULT ON
+	DESCRIPTION "Enables protocol extensions" DISABLED TORRENT_DISABLE_EXTENSIONS)
+target_optional_compile_definitions(torrent-rasterbar PUBLIC FEATURE NAME i2p DEFAULT ON
+	DESCRIPTION "build with I2P support" DISABLED TORRENT_USE_I2P=0)
+target_optional_compile_definitions(torrent-rasterbar PUBLIC FEATURE NAME ipv6 DEFAULT ON
+	DESCRIPTION "build with IPv6 support" DISABLED TORRENT_USE_IPV6=0)
+target_optional_compile_definitions(torrent-rasterbar PUBLIC FEATURE NAME logging DEFAULT ON
+	DESCRIPTION "build with logging" DISABLED TORRENT_DISABLE_LOGGING)
+
+find_public_dependency(Iconv)
```

Comment:
Maybe this call has to be made only on UNIX? From the other hand MinGW might use libiconv, if I'm not mistaken. `if (NOT MSVC)` perhaps?

Reply:
In theory I think you could use iconv even with MSVC, but I'm not sure why you would want to since the Windows SDK provides the same functionality. Maybe make the dep `OPTIONAL` instead of `RECOMMENDED` on MSVC.

Reply:
OK.